sdk/authorization/mgmt/src/main/java/com/azure/resourcemanager/authorization/implementation/ActiveDirectoryApplicationsImpl.java

```diff
@@ -93,10 +91,13 @@ public ActiveDirectoryApplication getByName(String spn) {
                 Mono
                     .defer(
                         () -> {
-                            UUID.fromString(trimmed);
-                            return inner().listAsync(String.format("appId eq '%s'", trimmed)).singleOrEmpty();
+                            try {
+                                UUID.fromString(trimmed);
+                                return inner().listAsync(String.format("appId eq '%s'", trimmed)).singleOrEmpty();
+                            } catch (IllegalArgumentException e) {
+                                return Mono.empty();
+                            }
```

Comment:
why does it need the `try`? When will it throw `IllegalArgumentException`?

Reply:
I think it is for `UUID.fromString`. So if input is not UUID, this part will be skipped.